services/121-service/src/payments/fsp-integration/nedbank/nedbank-api.service.spec.ts

```diff
@@ -0,0 +1,273 @@
+import { AxiosResponse } from '@nestjs/terminus/dist/health-indicator/http/axios.interfaces';
+import { Test, TestingModule } from '@nestjs/testing';
+import * as https from 'https';
+
+import { CreateOrderResponseNedbankDto } from '@121-service/src/payments/fsp-integration/nedbank/dtos/nedbank-api/create-order-response-nedbank.dto';
+import { ErrorReponseNedbankDto } from '@121-service/src/payments/fsp-integration/nedbank/dtos/nedbank-api/error-reponse-nedbank.dto';
+import { GetOrderResponseNedbankDto } from '@121-service/src/payments/fsp-integration/nedbank/dtos/nedbank-api/get-order-reponse-nedbank.dto';
+import { NedbankVoucherStatus } from '@121-service/src/payments/fsp-integration/nedbank/enums/nedbank-voucher-status.enum';
+import { NedbankError } from '@121-service/src/payments/fsp-integration/nedbank/errors/nedbank.error';
+import { NedbankApiService } from '@121-service/src/payments/fsp-integration/nedbank/nedbank-api.service';
+import { CustomHttpService } from '@121-service/src/shared/services/custom-http.service';
+import { registrationNedbank } from '@121-service/test/registrations/pagination/pagination-data';
+
+const amount = 250;
+const orderCreateReference = '123456'; // This is a uuid generated deterministically, so we can use a fixed value
+
+jest.mock('@121-service/src/shared/services/custom-http.service');
+
+describe('NedbankApiService', () => {
+  let service: NedbankApiService;
+  let httpService: CustomHttpService;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [NedbankApiService, CustomHttpService],
+    }).compile();
+
+    service = module.get<NedbankApiService>(NedbankApiService);
+    service.httpsAgent = {} as https.Agent;
+    httpService = module.get<CustomHttpService>(CustomHttpService);
+  });
+
+  describe('createOrder', () => {
+    it('should create an order successfully', async () => {
+      const response: AxiosResponse<CreateOrderResponseNedbankDto> = {
+        data: {
+          Data: {
+            OrderId: '',
+            Status: NedbankVoucherStatus.PENDING,
```

Comment:
Is this the only status that can be returned from createOrder()? If so, great. If not, unit test them as well?